pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -370,25 +372,32 @@ func (lb *ListenerBuilder) patchOneListener(l *listener.Listener, ctx networking
 		return nil
 	}
 	tempArray := []*listener.Listener{l}
-	tempArray = envoyfilter.ApplyListenerPatches(ctx, lb.node, lb.push, tempArray, true)
+	tempArray = envoyfilter.ApplyListenerPatches(ctx, lb.node, lb.push, lb.envoyFilterWrapper, tempArray, true)
 	// temp array will either be empty [if virtual listener was removed] or will have a modified listener
 	if len(tempArray) == 0 {
 		return nil
 	}
 	return tempArray[0]
 }
+
 func (lb *ListenerBuilder) patchListeners() {
+	lb.envoyFilterWrapper = lb.push.EnvoyFilters(lb.node)
```

Comment:
Why do we need it in the builder? It seems like we only use it locally in this function and patchOneListener so we could just pass it to patchOneListener possibly?

Reply:
buildHTTPProxyListener uses patchOneListener - that is why I added to ListenerBuilder